appbuilder/project/project-base.ts

```diff
@@ -40,5 +62,92 @@ export class Project implements Project.IProjectBase {
 	}
 
 	public startPackageActivity = startPackageActivityNames[TARGET_FRAMEWORK_IDENTIFIERS.Cordova.toLowerCase()];
+
+	public get hasBuildConfigurations(): boolean {
+		return this._hasBuildConfigurations;
+	}
+
+	public get projectInformation(): Project.IProjectInformation {
+		return {
+			projectData: this.projectData,
+			configurationSpecificData: this.configurationSpecificData,
+			hasBuildConfigurations: this.hasBuildConfigurations,
+			configurations: Object.keys(this.configurationSpecificData)
+		};
+	}
+
+	protected abstract validate(): void;
+	protected abstract saveProjectIfNeeded(): void;
+
+	protected readProjectData(): IFuture<void> {
+		return (() => {
+			let projectDir = this.getProjectDir().wait();
+			this.setShouldSaveProject(false);
+			if(projectDir) {
+				let projectFilePath = path.join(projectDir, this.$projectConstants.PROJECT_FILE);
+				try {
+					let data = this.$fs.readJson(projectFilePath).wait();
+					if(data.projectVersion && data.projectVersion.toString() !== "1") {
+						this.$errors.fail("FUTURE_PROJECT_VER");
+					}
+
+					if(!_.has(data, "Framework")) {
+						if(_.has(data, "projectType")) {
+							data["Framework"] = data["projectType"];
+							delete data["projectType"];
+						} else {
+							data["Framework"] = TARGET_FRAMEWORK_IDENTIFIERS.Cordova;
+						}
+
+						this.setShouldSaveProject(true);
+					}
+
+					this.projectData = data;
+
+					this.validate();
+
+					let debugProjectFile = path.join(projectDir, this.$projectConstants.DEBUG_PROJECT_FILE_NAME);
+					if(this.$options.debug && !this.$fs.exists(debugProjectFile).wait()) {
+						this.$fs.writeJson(debugProjectFile, {}).wait();
+					}
+
+					let releaseProjectFile = path.join(projectDir, this.$projectConstants.RELEASE_PROJECT_FILE_NAME);
+					if(this.$options.release && !this.$fs.exists(releaseProjectFile).wait()) {
+						this.$fs.writeJson(releaseProjectFile, {}).wait();
+					}
+
+					let allProjectFiles = this.$fs.enumerateFilesInDirectorySync(projectDir, (file: string, stat: IFsStats) => {
+						return ProjectBase.CONFIGURATION_FILE_SEARCH_PATTERN.test(file);
+					});
+
+					_.each(allProjectFiles, (configProjectFile: string) => {
+						let configMatch = path.basename(configProjectFile).match(ProjectBase.CONFIGURATION_FROM_FILE_NAME_REGEX);
+						if(configMatch && configMatch.length > 1) {
+							let configurationName = configMatch[1];
+							let configProjectContent = this.$fs.readJson(configProjectFile).wait();
+							this.configurationSpecificData[configurationName.toLowerCase()] = <any>_.merge(JSON.parse(JSON.stringify(this.projectData)), configProjectContent);
+							this._hasBuildConfigurations = true;
+						}
+					});
+				} catch(err) {
+					if(err === "FUTURE_PROJECT_VER") {
+						this.$errors.fail({
+							formatStr: "This project is created by a newer version of AppBuilder. Upgrade AppBuilder CLI to work with it.",
+							suppressCommandHelp: true
+						});
+					}
+					this.$errors.fail({
```

Comment:
actually it doesn't display any help because `suppressCommandHelp` is passed as `true`, however I'll still rewrite it as a call to `failWithoutHelp`